Fix macro casts in icProfileHeader.h

kateniolet opened the pull request:
Fix typos in various macros in icProfileHeader.h for enum bounds checking

xsscx commented:
Hello!

1. Can you confirm that `icMaxToneFunctuionSignature` is correct?

`icMaxToneFunctuionSignature` potentially may want to be changed to `icMaxToneFunctionSignature`

2. Can you provide reproduction steps to Compile for this PR?

Example
-
Compile Instructions based on Commit [f891074a0f1c9d61a3dfa53749265f8c14ed4ee6](https://github.com/InternationalColorConsortium/DemoIccMAX/tree/f891074a0f1c9d61a3dfa53749265f8c14ed4ee6)

XNU & Ubuntu
-
```
cd /tmp
wget https://github.com/InternationalColorConsortium/DemoIccMAX/archive/f891074a0f1c9d61a3dfa53749265f8c14ed4ee6.zip
unzip f891074a0f1c9d61a3dfa53749265f8c14ed4ee6.zip
cd DemoIccMAX-f891074a0f1c9d61a3dfa53749265f8c14ed4ee6/Build
cmake -DCMAKE_INSTALL_PREFIX=$HOME/.local -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS="-g -fsanitize=address,undefined -fno-omit-frame-pointer -Wall" -Wno-dev  Cmake
make -j$(nproc) 2>&1 | grep 'error:'
```
Result: PASS

xsscx reviewed (changes requested):
Hello @kateniolet 

1. Can you confirm that `icMaxToneFunctuionSignature` is correct?

`icMaxToneFunctuionSignature` potentially may want to be changed to `icMaxToneFunctionSignature`

2. Can you provide reproduction steps to Compile for this PR?

Example
-
Compile Instructions based on Commit [f891074a0f1c9d61a3dfa53749265f8c14ed4ee6](https://github.com/InternationalColorConsortium/DemoIccMAX/tree/f891074a0f1c9d61a3dfa53749265f8c14ed4ee6)

XNU & Ubuntu
-
```
cd /tmp
wget https://github.com/InternationalColorConsortium/DemoIccMAX/archive/f891074a0f1c9d61a3dfa53749265f8c14ed4ee6.zip
unzip f891074a0f1c9d61a3dfa53749265f8c14ed4ee6.zip
cd DemoIccMAX-f891074a0f1c9d61a3dfa53749265f8c14ed4ee6/Build
cmake -DCMAKE_INSTALL_PREFIX=$HOME/.local -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS="-g -fsanitize=address,undefined -fno-omit-frame-pointer -Wall" -Wno-dev  Cmake
make -j$(nproc) 2>&1 | grep 'error:'
```
Result: PASS

kateniolet commented:
1. Good catch! Fixed to 'Function' and squashed into the commit.

the change of icMaxCurveElemSignature to icMaxToneFunctionSignature

was because icMaxCurveElemSignature appeared to have been copy-pasted by the original author into the file twice and the second time it was following the icToneFunctionSignature enum - as these are simply macros for referencing the highest value of their respective enums I renamed this second one to fit its conext.

2. I'm not sure I understand the question - compiling is the same as usual, nothing changes there.

xsscx reviewed (approved):
Summary of PR: Fix typos in various macros in icProfileHeader.h for enum bounds checking

Reviewer: David Hoyt
Date: 05 July 2024
Merged by @xsscx

xsscx commented:
For Documentation, I used the following yaml to test this PR using PR 66 which is the currrent Last Known Good Commit for Compile using:
```
name: C/C++ CI

on:
  push:
    branches: [ "master" ]
  pull_request:
    branches: [ "master" ]
  workflow_dispatch:

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v4
      with:
        ref: f891074a0f1c9d61a3dfa53749265f8c14ed4ee6

    - name: make
      run: |
        cmake -DCMAKE_INSTALL_PREFIX=$HOME/.local -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS="-g -fsanitize=address,undefined -fno-omit-frame-pointer -Wall" -Wno-dev Build/Cmake
        make -j10
```
Result: PASS
